apps/insights/src/hooks/pyth-pro-demo/use-http-datastream.ts

```diff
@@ -0,0 +1,255 @@
+/* eslint-disable @typescript-eslint/no-unnecessary-condition */
+import type { Nullish } from "@pythnetwork/shared-lib/types";
+import { isNullOrUndefined, wait } from "@pythnetwork/shared-lib/util";
+import { useIsMounted } from "@react-hookz/web";
+import { useEffect, useRef, useState } from "react";
+
+import type {
+  UseDataStreamReturnType,
+  UseHttpDataStreamOpts,
+  UseHttpDataStreamReturnType,
+} from "./types";
+import { usePythProAppStateContext } from "../../context/pyth-pro-demo";
+import type {
+  AllAllowedSymbols,
+  AllDataSourcesType,
+  HistoricalDataResponseType,
+} from "../../schemas/pyth/pyth-pro-demo-schema";
+import {
+  HistoricalDataResponseSchema,
+  removeReplaySymbolSuffix,
+  ValidDateSchema,
+} from "../../schemas/pyth/pyth-pro-demo-schema";
+import { isReplayDataSource, isReplaySymbol } from "../../util/pyth-pro-demo";
+
+// 🚨 DEMO HACK ALERT: we pick some point into trading
+// to ensure all APIs have saturated query results
+const INITIAL_START_AT = new Date("2025-12-05T19:00:00.000Z");
+
+function getFetchHistoricalUrl(
+  datasource: Nullish<AllDataSourcesType>,
+  symbol: Nullish<AllAllowedSymbols>,
+  startAt: Date,
+) {
+  if (!isReplayDataSource(datasource) || !isReplaySymbol(symbol)) {
+    return "";
+  }
+  const startAtValidation = ValidDateSchema.safeParse(startAt);
+  if (startAtValidation.error) {
+    return "";
+  }
+
+  return `/api/pyth/get-pyth-feeds-demo-data/${datasource}/${removeReplaySymbolSuffix(symbol)}?startAt=${startAtValidation.data.toISOString()}`;
+}
+
+export async function fetchHistoricalData(
+  url: string,
+): Promise<HistoricalDataResponseType> {
+  const response = await fetch(url, {
+    method: "GET",
+  });
+  if (!response.ok) {
+    const errRes = (await response.json()) as { error: string };
```

Comment:
No need for the nasty typecasting if you just don't return errors as json which isn't useful anyways, just use text